api/lib/session/bloc/session_bloc.dart

```diff
@@ -311,9 +314,13 @@ class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
   }
 
   void _onGameEnded(OnGameEnded event, Emitter<SessionState> emit) {
+    final leaderboard = state.players.values.toList()
+      ..sort((a, b) => b.score.compareTo(a.score));
+    final newLeaderBoard = leaderboard.take(3);
     emit(
       state.copyWith(
         eventType: EventType.gameEnd,
+        leaderboard: newLeaderBoard.toList(),
```

Comment:
instead of creating new variable "newLeaderboard", directly pass leaderboard.take(3).toList()